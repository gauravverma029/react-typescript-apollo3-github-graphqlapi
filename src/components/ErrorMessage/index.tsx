import React from "react";

const ErrorMessage: React.FC = () => {
  return (
    <div className="px-4 py-3 flex bg-gray-100">
      <div className="flex-auto flex space-x-6">
        <div className="flex items-center justify-center text-gray-400">
          Something Wrong with Github Graphql Api. Please Try After sometime.
        </div>
      </div>
    </div>
  );
};
export default ErrorMessage;
