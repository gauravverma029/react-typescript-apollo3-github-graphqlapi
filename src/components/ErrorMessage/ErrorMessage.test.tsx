import React from "react";
import { cleanup, render } from "@testing-library/react";

import ErrorMessage from "./index";

afterEach(cleanup);

describe("ErrorMessage Component", () => {
  test("Should Render", async () => {
    const { asFragment } = render(<ErrorMessage />);
    const html = asFragment();
    expect(html).toMatchSnapshot();
  });

  test("Should Contain Text 'Something Wrong'", async () => {
    const { getByText } = render(<ErrorMessage />);
    getByText(/Something Wrong/i);
  });

  test("Should Contain div tag for text", async () => {
    const { getByText } = render(<ErrorMessage />);
    expect(getByText(/Something Wrong/i).tagName).toBe("DIV");
  });

  test("Should Contain ClassName", async () => {
    const { getByText } = render(<ErrorMessage />);
    expect(getByText(/Something Wrong/i).className).toBe(
      "flex items-center justify-center text-gray-400"
    );
  });
});
