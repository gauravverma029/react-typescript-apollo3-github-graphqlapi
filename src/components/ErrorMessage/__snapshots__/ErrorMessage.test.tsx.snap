// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Component Should Render 1`] = `
<DocumentFragment>
  <div
    class="px-4 py-3 flex bg-gray-100"
  >
    <div
      class="flex-auto flex space-x-6"
    >
      <div
        class="flex items-center justify-center text-gray-400"
      >
        Something Wrong with Github Graphql Api. Please Try After sometime.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
